src/v/cluster_link/replication/deps.h

```diff
@@ -72,14 +73,20 @@ class data_source {
      * Fetches some data, if any.
      */
     virtual ss::future<data> fetch_next(ss::abort_source&) = 0;
+
+    /**
+     * Returns the last seen log start offset from the data source
+     */
+    virtual kafka::offset last_seen_log_start_offset() = 0;
 };
 
 class data_source_factory {
 public:
     virtual ss::future<> start() = 0;
     virtual ss::future<> stop() noexcept = 0;
     virtual ~data_source_factory() = default;
-    virtual std::unique_ptr<data_source> make_source(const ::model::ntp&) = 0;
+    virtual std::unique_ptr<data_source>
+    make_source(const ::model::ntp&, ::model::timestamp) = 0;
```

Comment:
this feels very not natural to create a data source one need to pass in the timestamp